ibm/service/power/resource_ibm_pi_snapshot.go

```diff
@@ -245,33 +251,28 @@ func isWaitForPIInstanceSnapshotAvailable(ctx context.Context, client *st.IBMPIS
 	return stateConf.WaitForStateContext(ctx)
 }
 
-func isPIInstanceSnapshotRefreshFunc(client *st.IBMPISnapshotClient, id string) resource.StateRefreshFunc {
+func isPIInstanceSnapshotRefreshFunc(client *instance.IBMPISnapshotClient, id string) retry.StateRefreshFunc {
 	return func() (interface{}, string, error) {
-
 		snapshotInfo, err := client.Get(id)
 		if err != nil {
 			return nil, "", err
 		}
 
-		//if pvm.Health.Status == helpers.PIInstanceHealthOk {
-		if snapshotInfo.Status == "available" && snapshotInfo.PercentComplete == 100 {
+		if snapshotInfo.Status == State_Available && snapshotInfo.PercentComplete == 100 {
 			log.Printf("The snapshot is now available")
-			return snapshotInfo, "available", nil
+			return snapshotInfo, State_Available, nil
 
 		}
-		return snapshotInfo, "in_progress", nil
+		return snapshotInfo, State_InProgress, nil
 	}
 }
 
-// Delete Snapshot
-
-func isWaitForPIInstanceSnapshotDeleted(ctx context.Context, client *st.IBMPISnapshotClient, id string, timeout time.Duration) (interface{}, error) {
-
+func isWaitForPIInstanceSnapshotDeleted(ctx context.Context, client *instance.IBMPISnapshotClient, id string, timeout time.Duration) (interface{}, error) {
 	log.Printf("Waiting for (%s) to be deleted.", id)
 
-	stateConf := &resource.StateChangeConf{
-		Pending:    []string{"retry", helpers.PIInstanceDeleting},
-		Target:     []string{"Not Found"},
+	stateConf := &retry.StateChangeConf{
+		Pending:    []string{State_Retry, helpers.PIInstanceDeleting},
```

Comment:
Could you move this the constant file? ` helpers.PIInstanceDeleting`

Reply:
Good catch